File: backend/app/schemas/customer.py
from typing import Optional

from pydantic import BaseModel, ConfigDict, EmailStr


class CustomerBase(BaseModel):
    name: str
    email: EmailStr
    phone: Optional[str] = None
    billing_address: Optional[str] = None
    shipping_address: Optional[str] = None
    notes: Optional[str] = None


class CustomerCreate(CustomerBase):
    pass


class CustomerUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    billing_address: Optional[str] = None
    shipping_address: Optional[str] = None
    notes: Optional[str] = None
    is_active: Optional[bool] = None


class CustomerOut(CustomerBase):
    id: int
    is_active: bool = True

    model_config = ConfigDict(from_attributes=True)

File: frontend/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  email: string;
  phone: string | null;
  billing_address: string | null;
  shipping_address: string | null;
  notes: string | null;
  is_active: boolean;
}

export interface CustomerCreate {
  name: string;
  email: string;
  phone?: string | null;
  billing_address?: string | null;
  shipping_address?: string | null;
  notes?: string | null;
}

export interface CustomerUpdate {
  name?: string;
  email?: string;
  phone?: string | null;
  billing_address?: string | null;
  shipping_address?: string | null;
  notes?: string | null;
  is_active?: boolean;
}

